File: src/leibniz.rs
use std::fs::File;
use std::io::prelude::*;

fn main() {
    let mut file = File::open("./rounds.txt").expect("file not found");

    let mut contents = String::new();
    file.read_to_string(&mut contents)
        .expect("something went wrong reading the file");

    let rounds = contents.trim().parse::<u32>().unwrap() + 2;

    let mut pi: f64 = 1.0;
    (2..rounds).foreach(|i| {
        let x = -1.0f64 + (2.0 * (i & 0x1) as f64);
        pi += x / (2 * i - 1) as f64;
    });

    println!("{:.16}", pi * 4.0);
}

File: src/leibniz.dart
import 'dart:io';

void main() {
  final rounds = int.parse(File('rounds.txt').readAsStringSync().trim());

  double pi = 1.0;
  double x = 1.0;

  for (int i = 2; i <= rounds + 1; i++) {
    x = -x;
    pi += x / (2 * i - 1);
  }

  pi *= 4.0;
  print(pi.toStringAsFixed(16));
}
